src/components/TaskRunListItem.vue

```diff
@@ -1,43 +1,67 @@
 <template>
-  <StateListItem v-model:selected="model" v-bind="{ selectable, value, tags, stateType }" class="task-run-list-item">
-    <template #name>
-      <div class="task-run-list-item__breadcrumbs">
-        <template v-if="taskRun.flowRunId && !isFlowRunRoute && flowRun">
-          <p-link :to="routes.flowRun(taskRun.flowRunId)">
-            {{ flowRun.name }}
+  <div ref="el" class="task-run-list-item">
+    <StateListItem v-model:selected="model" v-bind="{ selectable, value, tags, stateType }">
+      <template #name>
+        <div class="task-run-list-item__breadcrumbs">
+          <template v-if="showFlow && flow">
+            <p-link class="task-run-list-item__flow-name" :to="routes.flow(flow.id)">
+              {{ flow.name }}
+            </p-link>
+            <p-icon icon="ChevronRightIcon" size="small" />
+          </template>
+
+          <template v-if="showFlowRun && flowRun">
+            <p-link :to="routes.flowRun(flowRun.id)">
+              {{ flowRun.name }}
+            </p-link>
+            <p-icon icon="ChevronRightIcon" size="small" />
+          </template>
+
+          <p-link :to="routes.taskRun(taskRun.id)">
+            <span>{{ taskRun.name }}</span>
           </p-link>
-          <p-icon icon="ChevronRightIcon" size="small" />
-        </template>
-
-        <p-link :to="routes.taskRun(taskRun.id)">
-          <span>{{ taskRun.name }}</span>
-        </p-link>
-      </div>
-    </template>
-    <template #meta>
-      <StateBadge :state="taskRun.state" />
-      <p-icon-text icon="ClockIcon">
-        {{ secondsToApproximateString(taskRun.duration) }}
-      </p-icon-text>
-      <p-icon-text class="flow-run-date-icon-text" icon="CalendarIcon">
-        <template v-if="taskRun.startTime">
-          {{ formatDateTimeNumeric(taskRun.startTime) }}
-        </template>
-        <template v-else-if="taskRun.expectedStartTime">
-          {{ formatDateTimeNumeric(taskRun.expectedStartTime) }}
-        </template>
-      </p-icon-text>
-    </template>
-  </StateListItem>
+        </div>
+      </template>
+      <template #meta>
+        <StateBadge :state="taskRun.state" />
+        <p-icon-text icon="ClockIcon">
+          {{ secondsToApproximateString(taskRun.duration) }}
+        </p-icon-text>
+        <p-icon-text class="flow-run-date-icon-text" icon="CalendarIcon">
+          <template v-if="taskRun.startTime">
+            {{ formatDateTimeNumeric(taskRun.startTime) }}
+          </template>
+          <template v-else-if="taskRun.expectedStartTime">
+            {{ formatDateTimeNumeric(taskRun.expectedStartTime) }}
+          </template>
+        </p-icon-text>
+      </template>
+
+      <template v-if="showFlowRun && visible" #relationships>
```

Comment:
can u eli5 what the intersection observer and visible toggle here are solving? 

is this like - a list of task run list items are rendered, some might be offscreen b/c scroll and for those we want to lazily load this slot?

Reply:
Yup exactly - these make a bunch of secondary calls to grab related items based on the ids/names passed here; we don't want to both making those calls if the user can't even see the list item.